src/decoder.rs

```diff
@@ -8,34 +8,56 @@ use nom::bytes::complete::{take, take_till};
 use nom::combinator::{map, map_parser};
 use nom::multi::many0;
 use nom::number::complete::{be_f32, be_f64, be_i32, be_i64, be_u32};
-use nom::{IResult,combinator::map_res,sequence::tuple};
+use nom::{IResult,combinator::map_res,sequence::tuple,Err};
+use OscError::BadPacket;
```

Comment:
For consistency we should remove this import and qualify the errors as `OscError::BadPacket`, this way it's also clear where the error type originates from.